crates/wit-parser/src/resolve.rs

```diff
@@ -341,13 +352,31 @@ impl Resolve {
     /// as [`Resolve::push_path`].
     ///
     /// Any dependency resolution error or otherwise world-elaboration error
-    /// will be returned here. If successful a package identifier is returned
+    /// will be returned here, if successful a package identifier is returned
     /// which corresponds to the package that was just inserted.
     pub fn push(&mut self, mut unresolved: UnresolvedPackage) -> Result<PackageId> {
         let source_map = mem::take(&mut unresolved.source_map);
         source_map.rewrite_error(|| Remap::default().append(self, unresolved))
     }
 
+    /// Appends new [`UnresolvedPackage`]s to this [`Resolve`], creating a
+    /// fully resolved package with no dangling references.
+    ///
+    /// The `deps` argument indicates that the named dependencies in
+    /// `unresolved` to packages are resolved by the mapping specified.
+    ///
+    /// Any dependency resolution error or otherwise world-elaboration error
+    /// will be returned here, if successful a package identifier is returned
+    /// which corresponds to the package that was just inserted.
+    pub fn append(&mut self, unresolved: Vec<UnresolvedPackage>) -> Result<Vec<PackageId>> {
+        let mut ids = Vec::new();
+        for mut pkg in unresolved {
+            let source_map = mem::take(&mut pkg.source_map);
+            ids.push(source_map.rewrite_error(|| Remap::default().append(self, pkg))?);
+        }
+        Ok(ids)
+    }
```

Comment:
Since this is "just" a simple loop calling `push`, could this be left out instead?

Either that or this might perhaps need some extra refactoring to make it work. Namely if this method take a list of unresolved packages then I'd expect it to automatically topologically sort the packages for insertion into `Resolve` so it's not on the user to manually sort the packages.

One idea perhaps is to have `push_dir` bottom out here where the topo-sort performed there is now performed here instead?

Reply:
Done, this is now topologically sorted.